### PR TITLE
Feature/michael settimeout to battle results

### PR DESCRIPTION
**Pull request by mklemersson**
- Add the scroll bar
- Add time out to show the battle rounds

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -16,5 +16,9 @@ require('laravel-elixir-vue-2');
 elixir(mix => {
     mix.sass('app.scss')
        .webpack('app.js')
-       .copy('node_modules/bootstrap-sass/assets/fonts/bootstrap/','public/fonts/bootstrap');;
+       .scripts([
+           '../../../node_modules/malihu-custom-scrollbar-plugin/jquery.mCustomScrollbar.js'
```
Comment (edddieee) [inline gulpfile.js]: I think that would be a good ideia to require this file into `resources/assets/app.js` and not here.

File: resources/assets/sass/app.scss [hunk 2]
```diff
@@ -8,6 +8,9 @@
 // Bootstrap
 @import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap";
 
+// Custom Scroll Bar
+@import "customscrollbar";
```
Comment (edddieee) [inline resources/assets/sass/app.scss]: You should to import this file from `node_modules` folder, like this:

`@import "node_modules/malihu-custom-scrollbar-plugin/jquery.mCustomScrollbar.css"`

File: resources/assets/sass/_customscrollbar.scss [hunk 3]
```diff
@@ -0,0 +1,1267 @@
+/*
```
Comment (edddieee) [inline resources/assets/sass/_customscrollbar.scss]: I think you should remove this file. You have already imported it.
https://github.com/timongorpg/timongo/pull/37/files#diff-1dee3aa6a79c9f1c3bf0abca2c6fd30fR12

File: resources/views/battle-results.blade.php [hunk 4]
```diff
@@ -44,16 +44,17 @@
...
-                    @if($message['hero'])
-                        <img src="/img/icons/attack.png" class="battle-icon" alt="">
-                        </object>
-                    @endif
-
-                    {!! $message['message'] !!}
-                </div>
-            @endforeach
+            <div id="battle-results" class="content mCustomScrollbar">
+                <!-- @foreach(array_get($log, 'fight') as $message)
```
Comment (edddieee) [inline resources/views/battle-results.blade.php]: If this code isn't necessary, you should remove it.

File: gulpfile.js (hunk 1 above)
Comment (edddieee) [reply gulpfile.js]: I think we should minify the output, before merge this PR into master.
`npm run production`

**Review by edddieee — CHANGES_REQUESTED** [review]

File: resources/assets/sass/app.scss (hunk 2 above)
Comment (mklemersson) [reply resources/assets/sass/app.scss]: I think that we can't import css into scss, no?

Comment (edddieee) [reply resources/assets/sass/app.scss]: You can import css normally. 😄

**Comment by mklemersson** [comment]
I did it but the code wasn't imported :(



Atenciosamente,

Michael Klemersson - http://www.michaelklemersson.com

https://br.linkedin.com/in/michaelklemersson

2017-03-22 10:27 GMT-03:00 Edgar Yonemura <notifications@github.com>:

**Comment by mklemersson** [comment]
PR updated :), code is minified and scroll bar is now scrolling to the last battle event :+1:

File: resources/assets/js/app.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ require('./moment');
 require('./blink-title');
 require('./form-functions');
 
-require('malihu-custom-scrollbar-plugin')
+require('../../../node_modules/malihu-custom-scrollbar-plugin/jquery.mCustomScrollbar.concat.min')
```
Comment (edddieee) [inline resources/assets/js/app.js]: You forgot the `;`. 
What do you think to move this require to `bootstrap.js`?

**Review by edddieee — APPROVED** [review]

File: resources/assets/js/app.js (hunk 5 above)
Comment (mklemersson) [reply resources/assets/js/app.js]: next PR :)

**Comment by hudsonpereira** [comment]
I'll check this PR tonight. @eddiecode has good thoughts about it.

**Comment by edddieee** [comment]
More one thing that I forgot to say, but I think that would be a good idea to fixup or squash the last two commits. Let me know if you don't know how to do this. cc @MichaelKlemersson

**Comment by mklemersson** [comment]
I don't know @eddiecode how to squash for a while :(

**Comment by edddieee** [comment]
It's a very simple thing. Both squash and fixup are rebase commands to merge two or more commits. The differences is that squash don't remove the commit message.

To do a rebase you need to run the follow commands:

```
# Run rebase interactive command and get the last three commits (HEAD~3)
$ git rebase -i HEAD~3
```

git will open your text editor, probably vim/vi if you are using Linux or Mac.

You will see something like that:
```
pick a3240d1 add slim scroll to battle results
pick 31eb90c add scroll bar and timeout to show battle results
pick 3c18ece update import scrollbar plugin js and css

# Rebase a3240d1..3c18ece onto 3c18ece
#
# Commands:
#  p, pick = use commit
#  r, reword = use commit, but edit the commit message
#  e, edit = use commit, but stop for amending
#  s, squash = use commit, but meld into previous commit
#  f, fixup = like "squash", but discard this commit's log message
#  x, exec = run command (the rest of the line) using shell
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

And then you can squash(merge) the last two commits:

```
pick a3240d1 add slim scroll to battle results
pick 31eb90c add scroll bar and timeout to show battle results
squash 3c18ece update import scrollbar plugin js and css
```

Save the file, and the rebase will be done. 😄 

cc @MichaelKlemersson

**Comment by mklemersson** [comment]
@eddiecode I did it, so what now?

**Comment by edddieee** [comment]
@MichaelKlemersson run `$ git log --oneline` and check if your commits were merged.
And then you just need to push your branch again: `$ git push -f your_remote_repository_name your_branch_name`

**Comment by mklemersson** [comment]
I did so, I think that is ready

**Comment by edddieee** [comment]
@MichaelKlemersson for me this PR it's ok. Can you check it later @hudsonpereira?

**Pull request merged**